File: OneAPI/Ruby/SendSMS/send_sms_with_options.rb
require 'uri'
require 'json'
require 'net/http'
require 'openssl'

# Your Engagement Cloud CPaaS settings
Username = "YOUR API USERNAME"
Password = "YOUR API PASSWORD"

puts ""
puts "Sending SMS using Engagement Cloud CPaaS and Ruby"
puts "-------------------------------------------------"

# Setup the conneciton object
url = URI("https://api-cpaas.dotdigital.com/cpaas/messages")
http = Net::HTTP.new(url.host, url.port)
http.use_ssl = true
http.verify_mode = OpenSSL::SSL::VERIFY_NONE # This should be removed and SSL certs validated for production environments

# Setup the HTTP request
request = Net::HTTP::Post.new(url)
request.basic_auth Username, Password
request["content-type"] = 'application/json'
request["accept"] = 'application/json'
request["cache-control"] = 'no-cache'

# Create the Engagement Cloud CPaaS request JSON
request.body = 
    "{
      \"body\": \"Your SMS message\",
      \"to\": {
          \"phoneNumber\": \"447123123123\"
        },
      \"channelOptions\": {
          \"sms\": {
              \"from\": \"EC CPaaS\",
              \"allowUnicode\": true
            }
        },
     \"rules\": [ \"sms\" ]
     }"

# Call the web service
puts ""
puts "Calling Engagement Cloud CPaaS..."
response = http.request(request)

puts ""
puts "Call returned status code: " + response.code
puts JSON.pretty_unparse(JSON.parse(response.read_body)) # Pretty print the JSON

File: OneAPI/Python/SendSMS/send_sms_with_options.py
import http.client
from base64 import b64encode
import json

print("")
print("Sending SMS using Engagement Cloud CPaaS and Python")
print("---------------------------------------------------")

# Your Engagement Cloud API user credentials
API_USERNAME = "YOUR API USERNAME"
API_PASSWORD = "YOUR API PASSWORD"

# Setup the http connection
conn = http.client.HTTPSConnection("api-cpaas.dotdigital.com")

# Create the basic auth header encoded username and password
userAndPass = b64encode(bytes(API_USERNAME + ":" + API_PASSWORD, "ascii")).decode("ascii")

# Construct the Comapi API request
myRequest = {
    "to": {
        "phoneNumber": "447123123123"
    },
    "body": "This is an SMS via Engagement Cloud CPaaS \"One\" API",
    "channelOptions":
    {
        "sms": {
            "from": "EC CPaaS",
            "allowUnicode": True
        }
    },
    "rules": ["sms"]
}

print("")
print("Request JSON: ")
print(json.dumps(myRequest, indent=2))

# Setup the http headers
headers = {
    'authorization' : 'Basic %s' %  userAndPass,
    'content-type': "application/json",
    'cache-control': "no-cache"
}

# Make the webservice request
print("")
print("Calling Engagement Cloud CPaaS...")
conn.request("POST", "/cpaas/messages",
             json.dumps(myRequest), headers)

res = conn.getresponse()
data = res.read()

print("")
print("Call returned status code: " + str(res.status))
print(json.dumps(json.loads(data.decode("utf-8")), indent=2)) # Pretty print the JSON
print("")